### PR TITLE
feat: API includes more fields to sort on during sessions list

### PR DESCRIPTION
**Pull request by aneojgurhem**

### CODE REVIEW

File: Common/src/ArmoniK.Core.Common.csproj [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 
   <ItemGroup>
-    <PackageReference Include="ArmoniK.Api.Core" Version="3.10.0" />
+    <PackageReference Include="ArmoniK.Api.Core" Version="3.11.0-feataddtaskoptionstosession.23.58da3e7" />
```
Comment (ddiakiteaneo) [inline Common/src/ArmoniK.Core.Common.csproj:29]: Can this API version go on main?

Comment (aneojgurhem) [reply Common/src/ArmoniK.Core.Common.csproj:29]: Not yet, we need to merge the new Api first

**Comment by aneojgurhem** [comment]
This PR lacks some tests on the usage of the new fields

**Comment by aneojgurhem** [comment]
I added the tests

File: Common/tests/TestBase/SessionTableTestBase.cs [hunk 2]
```diff
@@ -454,4 +443,38 @@ public async Task ListSessionAsyncNoFilterShouldSucceed()
                       res.Count);
     }
   }
+
+
+  [Test]
+  public async Task ListSessionAsyncTaskOptionsOptions()
```
Comment (dbrasseur-aneo) [inline Common/tests/TestBase/SessionTableTestBase.cs:449]: This tests the number of sessions, not that the sort was done correctly

Comment (aneojgurhem) [reply Common/tests/TestBase/SessionTableTestBase.cs:449]: Yes, that is the role of the database.

Comment (aneojgurhem) [reply Common/tests/TestBase/SessionTableTestBase.cs:449]: It mainly tests that the request is valid

**Review by dbrasseur-aneo — APPROVED** [review]

**Pull request merged**